src/frontend/src/api/CreateProjectService.ts

```diff
@@ -190,34 +190,37 @@ const GenerateProjectFilesService = (url: string, projectData: any, formUpload:
       let isAPISuccess = true;
       try {
         let response;
+
         const additional_entities: string[] =
           projectData?.additional_entities?.length > 0
             ? [projectData?.additional_entities?.[0]?.replaceAll(' ', '_')]
             : [];
+
         if (projectData.form_ways === 'custom_form') {
           // TODO move form upload to a separate service / endpoint?
           const generateApiFormData = new FormData();
           generateApiFormData.append('xlsform', formUpload);
+
           if (additional_entities?.length > 0) {
             generateApiFormData.append('additional_entities', additional_entities);
           }
+
           response = await axios.post(url, generateApiFormData, {
             headers: {
               'Content-Type': 'multipart/form-data',
             },
           });
         } else {
-          const generateApiFormData = new FormData();
-          generateApiFormData.append(
-            'additional_entities',
-            additional_entities?.length > 0 ? additional_entities : null,
-          );
-          response = await axios.post(url, generateApiFormData, {
+          const payload = {
+            additional_entities: additional_entities.length > 0 ? additional_entities : null,
+          };
+          response = await axios.post(url, payload, {
             headers: {
-              'Content-Type': 'multipart/form-data',
+              'Content-Type': 'application/json',
```

Comment:
Well done finding the problem!
It makes sense now 😄

But here I think form data was used to provide a file name, so that can give a name to the additional entity list.

Does this work correctly when using a json upload?

Reply:
Yes, this works correctly. :tada:

Reply:
The form data was used because of custom xlsform upload, we don't need form data only for the additional_entities list.